File: src/main/java/com/pengrad/telegrambot/model/Update.java
package com.pengrad.telegrambot.model;

/**
 * stas
 * 8/4/15.
 */
public class Update {

    private Integer update_id;
    private Message message;
    private Message edited_message;
    private InlineQuery inline_query;
    private ChosenInlineResult chosen_inline_result;
    private CallbackQuery callback_query;

    public Integer updateId() {
        return update_id;
    }

    public Message message() {
        return message;
    }

    public Message editedMessage() {
        return edited_message;
    }

    public InlineQuery inlineQuery() {
        return inline_query;
    }

    public ChosenInlineResult chosenInlineResult() {
        return chosen_inline_result;
    }

    public CallbackQuery callbackQuery() {
        return callback_query;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Update update = (Update) o;

        if (update_id != null ? !update_id.equals(update.update_id) : update.update_id != null) return false;
        if (message != null ? !message.equals(update.message) : update.message != null) return false;
        if (edited_message != null ? !edited_message.equals(update.edited_message) : update.edited_message != null)
            return false;
        if (inline_query != null ? !inline_query.equals(update.inline_query) : update.inline_query != null) return false;
        if (chosen_inline_result != null ? !chosen_inline_result.equals(update.chosen_inline_result) : update.chosen_inline_result != null)
            return false;
        return callback_query != null ? callback_query.equals(update.callback_query) : update.callback_query == null;

    }

    @Override
    public int hashCode() {
        return update_id != null ? update_id.hashCode() : 0;
    }

    @Override
    public String toString() {
        return "Update{" +
                "update_id=" + update_id +
                ", message=" + message +
                ", edited_message=" + edited_message +
                ", inline_query=" + inline_query +
                ", chosen_inline_result=" + chosen_inline_result +
                ", callback_query=" + callback_query +
                '}';
    }
}

File: library/src/main/java/com/pengrad/telegrambot/model/Update.kt
package com.pengrad.telegrambot.model

import java.io.Serializable

/**
 * stas
 * 8/4/15.
 */
data class Update(
        @get:JvmName("updateId") val update_id: Int? = null,
        @get:JvmName("message") val message: Message? = null,
        @get:JvmName("editedMessage") val edited_message: Message? = null,
        @get:JvmName("channelPost") val channel_post: Message? = null,
        @get:JvmName("editedChannelPost") val edited_channel_post: Message? = null,
        @get:JvmName("inlineQuery") val inline_query: InlineQuery? = null,
        @get:JvmName("chosenInlineResult") val chosen_inline_result: ChosenInlineResult? = null,
        @get:JvmName("callbackQuery") val callback_query: CallbackQuery? = null,
        @get:JvmName("shippingQuery") val shipping_query: ShippingQuery? = null,
        @get:JvmName("preCheckoutQuery") val pre_checkout_query: PreCheckoutQuery? = null,
        @get:JvmName("poll") val poll: Poll? = null,
        @get:JvmName("pollAnswer") val poll_answer: PollAnswer? = null
) : Serializable {
    companion object {
        private const val serialVersionUID = 0L
    }
}